src/platform/qpg6100/BLEManagerImpl.cpp

```diff
@@ -405,67 +389,270 @@ void BLEManagerImpl::DriveBLEState(intptr_t arg)
     sInstance.DriveBLEState();
 }
 
-CHIP_ERROR BLEManagerImpl::HandleGAPConnect(const ChipDeviceEvent * event)
+void BLEManagerImpl::HandleTXCharRead(uint16_t connId, uint16_t handle, uint8_t operation, uint16_t offset,
+                                      qvCHIP_Ble_Attr_t * pAttr)
 {
-    ChipLogProgress(DeviceLayer, "BLE GAP connection established");
+    uint8_t rsp = 0;
 
-    // TODO - handle platform specific information
+    ChipLogProgress(DeviceLayer, "Read request received for CHIPoBLE TX characteristic (con %u)", connId);
 
-    // Track the number of active GAP connections.
-    mNumGAPCons++;
+    // Send a zero-length response.
+    qvCHIP_TxData(connId, handle, 0, &rsp);
+}
 
-    // Update advertising state on connection
-    SetFlag(mFlags, kFlag_AdvertisingRefreshNeeded);
-    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+void BLEManagerImpl::HandleRXCharWrite(uint16_t connId, uint16_t handle, uint8_t operation, uint16_t offset, uint16_t len,
+                                       uint8_t * pValue, qvCHIP_Ble_Attr_t * pAttr)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    PacketBuffer * buf = NULL;
```

Comment:
nit: I believe we currently mostly use `nullptr` in CHIP

Reply:
agreed.
Note - ironically - only search hits on NULL in https://github.com/project-chip/connectedhomeip/blob/master/docs/style/coding/CODING_STYLE_GUIDE.adoc

Reply:
That is hilarious :rofl: